client/state/current-user/email-verification/reducer.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { createReducer } from 'state/utils';
+// import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_VERIFY_REQUEST,
+	EMAIL_VERIFY_REQUEST_SUCCESS,
+	EMAIL_VERIFY_REQUEST_FAILURE,
+	EMAIL_VERIFY_STATE_RESET,
+} from 'state/action-types';
+
+export default createReducer( {}, {
+	[ EMAIL_VERIFY_REQUEST ]: () => ( {
+		pendingRequest: true,
+		emailSent: false,
+		error: false,
```

Comment:
> We likely won't need to keep track of pending state if we use the data-layer

@gwwar I thought about it, and I didn't find a way to remove the pending state. Here are the various states the UI cares about:

1) Initial: no request has been made recently
2) Pending: to show the loading spinner
3) Email successful: to show success
4) Unsuccessful send: to show error message (not used in `VerifyEmailDialog`, but used in other components)

Am I missing a way this could be simplified now that I'm using the data layer?